Final configuration for Prusa Mini and Prusa Mini Bear

jfestrada opened the pull request:
This Pull Request include all the specific configurations for the Prusa Mini and the Bear variant. We had to define some configurations outside of the printers/prusa-mini folder. 
Here are the details of what is defined outside of the prusa-mini folder: 
- extruders/mk3s.cfg --> is the default extruder for the bear variant of the Prusa Mini
- extruders/prusa-mini.cfg --> is the default extruder for the Prusa Mini (bowden)
- extruders/tl-31.cfg --> is the definition of the extruder motor from trianglelab which include the 3:1 gear ratio
- scripts/install-prusa-mini-config.sh --> just for copy the template for the prusa mini to the printer.cfg
- steppers/ldo/42sth40-1684ac/2209/24v-0.4a-x.cfg  --> 0.4a configuration for x motor
- steppers/ldo/42sth40-1684ac/2209/24v-0.4a-y.cfg --> 0.4a configuration for y motor
- steppers/ldo/42sth40-1684ac/2209/24v-0.4a-z.cfg --> 0.4a configuration for z motor
- templates/prusa-mini-printer.template.cfg --> template configuration
- boards/prusa-buddy/config.cfg --> added the filament_sensor pin
- printers/initial-setup.cfg --> added the prusa mini initial setup
- scripts/compile-binaries.sh --> added the prusa buddy to the compilation list

![image](https://user-images.githubusercontent.com/7290345/208271708-1c603dcb-1da7-405b-a400-87f1d4d8cfe8.png)


https://user-images.githubusercontent.com/7290345/208271660-dc797a1b-37f6-4d30-8994-5476b811eecd.MP4

jfestrada committed:
final configuration for Prusa Mini a Prusa Mini Bear

jfestrada committed:
remove save_config section

jfestrada committed:
Fixed control tag for the heated_bed

jfestrada committed:
Fixed PID for extruder and heated bed

jfestrada committed:
Added z_offset to the template file

jfestrada committed:
Actualizar prusa-mini.cfg

Delete the pid from the prusa-mini.cfg

jfestrada committed:
Actualizar prusa-mini-printer.template.cfg

Added the pid for the heater bed to the template

miklschmidt commented:
Excellent work!

Is there a good reason to overriding the `LOAD_FILAMENT`, `UNLOAD_FILAMENT`, `PAUSE`, `RESUME` and `CANCEL_PRINT` macros? What are the RatOS macro's missing to work with the Prusa printers?

jfestrada committed:
Delete override of LOAD_FILAMENT, UNLOAD_FILAMENT, PAUSE, RESUME and CANCEL_PRINT

jfestrada commented:
> Excellent work!
> 
> Is there a good reason for overriding the `LOAD_FILAMENT`, `UNLOAD_FILAMENT`, `PAUSE`, `RESUME` and `CANCEL_PRINT` macros? What are the RatOS macro's missing to work with the Prusa printers?

You were right, that was because I started with the customization of a standard mainsail using as base the MK3S configuration. I have tested the printer behavior with the RatOS macros and is working fine.

jfestrada committed:
Merge branch 'v2.x-PrusaMini' of github.com:ComunidadMaker3D/CM3D-klipper-configuration into v2.x-PrusaMini

miklschmidt commented on steppers/ldo/42sth40-1684ac/2209/24v-0.4a-x.cfg:
```diff
@@ -0,0 +1,9 @@
+[tmc2209 stepper_x]
+uart_pin: board_uart_pin
+uart_address: 1
+sense_resistor: 0.22
```
These are board specific. Change uart_pin to `x_uart_pin` and move uart_address and sense_resistor to the prusa_buddy quirks filed i mentioned in another comment.

jfestrada replied:
Done, but I will test it before commit

jfestrada replied:
Tested and everything is ok

miklschmidt commented on steppers/ldo/42sth40-1684ac/2209/24v-0.4a-y.cfg:
```diff
@@ -0,0 +1,9 @@
+[tmc2209 stepper_y]
+uart_pin: board_uart_pin
+uart_address: 3
+sense_resistor: 0.22
```
These are board specific. Change uart_pin to `y_uart_pin` and move uart_address and sense_resistor to the prusa_buddy quirks filed i mentioned in another comment.

jfestrada replied:
Done

miklschmidt commented on steppers/ldo/42sth40-1684ac/2209/24v-0.4a-z.cfg:
```diff
@@ -0,0 +1,9 @@
+[tmc2209 stepper_z]
+uart_pin: board_uart_pin
+uart_address: 0
+sense_resistor: 0.30
```
These are board specific. Change uart_pin to `z_uart_pin` and move uart_address and sense_resistor to the prusa_buddy quirks filed i mentioned in another comment.

jfestrada replied:
Done

miklschmidt commented on templates/prusa-mini-printer.template.cfg:
```diff
@@ -0,0 +1,254 @@
+pid_kd = 932.049
+
+[respond]
+default_type: command
```
What is this for? Is it required for a display or something? the `respond` section is already present in printers/base.cfg, i'm fine with changing the type there if needed.

jfestrada replied:
Is not needed for the Mini because the display is not supported by klipper.

miklschmidt commented on templates/prusa-mini-printer.template.cfg:
```diff
@@ -0,0 +1,254 @@
+default_type: command
+
+[virtual_sdcard]
+path:/home/pi/printer_data/gcodes
```
remove this section, this is already handled by RatOS.

jfestrada replied:
Done

miklschmidt commented on templates/prusa-mini-printer.template.cfg:
```diff
@@ -0,0 +1,254 @@
+[pause_resume]
+recover_velocity: 50.
+
+[exclude_object]
```
Remove this section, it's not enabled by default because it requires a lot of processing, can be a problem with older raspberry pi's. It's something the user should enable themselves for this reason. (it doesn't work anyway without a similar section in moonraker.conf)

jfestrada replied:
It was there only for test purposes

miklschmidt commented on templates/prusa-mini-printer.template.cfg:
```diff
@@ -0,0 +1,254 @@
+#   be provided.
+#on_error_gcode:
+#   A list of G-Code commands to execute when an error is reported.
+
```
We're missing a USER OVERRIDES section at the end here.

jfestrada replied:
Already added

miklschmidt commented on templates/prusa-mini-printer.template.cfg:
```diff
@@ -0,0 +1,254 @@
+# Home, get position, throw around toolhead, home again.
+# If MCU stepper positions (first line in GET_POSITION) are greater than a full step different (your number of microsteps), then skipping occured.
+# We only measure to a full step to accomodate for endstop variance.
+# Example: TEST_SPEED SPEED=300 ACCEL=5000 ITERATIONS=10
```
This comment section should probably be removed. There is no `TEST_SPEED` macro either.

jfestrada replied:
Same it was used to determine the propper velocity, over 160mm/s the Y belt start to vibrate in a weird way.

miklschmidt commented on templates/prusa-mini-printer.template.cfg:
```diff
@@ -0,0 +1,254 @@
+# Example: TEST_SPEED SPEED=300 ACCEL=5000 ITERATIONS=10
+
+#[mcu USB-ADXL345]
+#serial: /dev/serial/by-id/usb-Klipper_stm32f103xe_36FFD7054858383038652443-if00
```
Remove this, it's not a thing in RatOS. We could add separate support for a USB adxl345 board if necessary.

jfestrada replied:
In the case of the prusa mini with the buddy board you must connect the ADLX345 sensor to the rpi, there is no possibility to connect to the board. So I added a new toolboard input-shaper-usb.cfg, the problem is that the user should override the pinout (maybe) and the mcu serial

miklschmidt replied:
You can connect it to the pi and use the rpi config: https://github.com/Rat-OS/RatOS-configuration/blob/v2.x/sensors/rpi-adxl345.cfg and https://github.com/Rat-OS/RatOS-configuration/blob/v2.x/boards/rpi/config.cfg. A usb controller used solely for ADXL345 is not how you connect an ADXL345 to the pi.

You'll want to have a config that sets
```
[resonance_tester]
accel_chip: adxl345 rpi
```

jfestrada replied:
I understand, so I have left the same configuration than for the v-minion

miklschmidt commented on templates/prusa-mini-printer.template.cfg:
```diff
@@ -0,0 +1,254 @@
+
+#[adxl345]
+#cs_pin: USB-ADXL345:PA4
+#spi_bus:spi1
```
Same with this. Remove.

jfestrada replied:
Done

miklschmidt commented on templates/prusa-mini-printer.template.cfg:
```diff
@@ -0,0 +1,254 @@
+#probe_points:
+#    90, 90, 120
+
+#   The path of the local directory on the host machine to look for
```
Remove all the following lines.

jfestrada replied:
Done

jfestrada commented on boards/prusa-buddy/config.cfg:
```diff
@@ -20,6 +20,8 @@ aliases:
   heater_bed_heating_pin=PB0,
   heater_bed_sensor_pin=PA4,
+#Filament sensor
+  filament_sensor=PB4,
```
Filament sensor pin renamed as suggested

jfestrada commented on printers/prusa-mini/macros.cfg:
```diff
@@ -0,0 +1,37 @@
+event_delay: 3.0
+pause_delay: 0.01
+
+[gcode_macro _M600]
```
Just waste, I have deleted the macro _M600

jfestrada commented on printers/prusa-mini/macros.cfg:
```diff
@@ -0,0 +1,37 @@
+# To override settings from this file, you can copy and paste the relevant
+# sections into your printer.cfg and change it there.
+
+[filament_switch_sensor filament_sensor]
```
New file for the filament switch generated and included on the prusa mini template.

jfestrada commented on printers/prusa-mini/tmc2209.cfg:
```diff
@@ -0,0 +1,21 @@
+
+[tmc2209 stepper_x]
+interpolate: False
+run_current: 0.5
```
Done

jfestrada commented on printers/prusa-mini/tmc2209.cfg:
```diff
@@ -0,0 +1,21 @@
+
+[tmc2209 stepper_y]
+interpolate: False
+run_current: 0.5
```
Done

jfestrada commented on printers/prusa-mini/tmc2209.cfg:
```diff
@@ -0,0 +1,21 @@
+
+[tmc2209 stepper_z]
+interpolate: False
+run_current: 0.5
```
Done

jfestrada commented on printers/prusa-mini/tmc2209.cfg:
```diff
@@ -0,0 +1,21 @@
+
+[tmc2209 extruder]
+interpolate: False
+uart_pin:  board_uart_pin
```
Done

jfestrada commented on printers/prusa-mini/steppers.cfg:
```diff
@@ -0,0 +1,33 @@
+step_pin: x_step_pin
+dir_pin: x_dir_pin
+enable_pin: !x_enable_pin
+microsteps: 16
```
Changed on X, Y and Z

jfestrada committed:
Requested changes for PR 93

jfestrada commented on printers/prusa-mini/prusa-mini.cfg:
```diff
@@ -0,0 +1,65 @@
+#   fan_speed 1.0: 100% PWM = 8000RPM (safe but loud)
+[heater_fan hotend_fan]
+pin: fan_toolhead_cooling_pin
+tachometer_pin: fan_toolhead_cooling_tachonmeter_pin
```
Done

jfestrada committed:
Update of the length distances for load/unload actions according with the Prusa Mini bowden tube.

jfestrada commented:
All the requested changes has been done and tested.

miklschmidt commented:
Excellent, looks like you just gotta merge v2.x and we're good to go :)

miklschmidt commented:
Oh and one last thing, due to not being able to connect an ADXL345 to the buddy board, i suggest we kill the input-shaper-xy config for now and make input-shaper-single.cfg use the rpi adxl345 for that.

jfestrada committed:
As the Buddy board does not have a clear instructions of how to connect
the ADLX345 directly to the board, the best approach is to connect it to
the RPI, this way there is no place to the error or mistakes.

input-shaper-single.cfg will have now the configuration to use the RPI
as host of the ADLX345.

jfestrada commented:
> Oh and one last thing, due to not being able to connect an ADXL345 to the buddy board, i suggest we kill the input-shaper-xy config for now and make input-shaper-single.cfg use the rpi adxl345 for that.

You are right, this way there is no place to the error/mistake. I have made the changes so the input-shaper-single.cfg have the configuration for the ADLX345 using the RPI mcu.

jfestrada committed:
Merge branch 'v2.x-PrusaMini' into v2.x

jfestrada committed:
Fixed the offset of the Prusa Mini Z probe, this was very close to the offset of the EVA3 because of that I didn't notice before

miklschmidt commented:
Excellent, now we just need a
```
git rebase v2.x
git push -f
```
And it should be good to merge :)

jfestrada commented:
> Excellent, now we just need a
> 
> ```
> git rebase v2.x
> git push -f
> ```
> 
> And it should be good to merge :)

Done

miklschmidt commented:
@jfestrada Do you have an actual model name for `extruders/tl-31.cfg` ?
